### PR TITLE
feat(internal): Make changelog JSON available to commit message body

### LINKED ISSUE #6337: Can't use changelog in commit body template
**What Renovate type are you using?**

Hosted Github App

**Describe the bug**

I'm trying to use #5966 to put the release notes into the commit body, but I can't find a workable method. There's a bunch of stuff tried here: https://github.com/mikebryant/renovate-test-changelogjson/blob/master/renovate.json, with the output here: https://github.com/mikebryant/renovate-test-changelogjson/pull/2/commits/ab443397dd07cbb8612a747375f134512ce1565d

In particular:
- `releases` key isn't present.
- `upgrades` key isn't present.
- `logJSON` key isn't present.

**Did you see anything helpful in debug logs?**

**To Reproduce**

See https://github.com/mikebryant/renovate-test-changelogjson/blob/master/renovate.json

**Additional context**

I think the first two issues are because [lib/workers/pr/index.ts] is where they're generated, and that's not before branch/commit creation happens?

I think the latter issue (logJSON) is because it's not an allowed field in [lib/util/template/index.ts]

I'm not sure what the desired route forward would be - should the upgrade / releases field creation be moved up into the [lib/workers/repository/index.ts], where logJSON creation is? Or should logJSON (and all of the nested fields) be an allowed field?

### PR DESCRIPTION
**Pull request by mikebryant**
I think I've kept the concurrent changelog fetching

Example of this working is here: https://github.com/mikebryant/renovate-test-changelogjson/pull/5/commits

Closes #6337

### CODE REVIEW

File: lib/workers/repository/changelog/index.ts [hunk 1]
```diff
@@ -7,10 +7,10 @@ async function embedChangelog(upgrade): Promise<void> {
 }
 
 // istanbul ignore next
-export async function embedChangelogs(branches): Promise<void> {
+export async function embedChangelogs(branchUpgrades): Promise<void> {
   const upgrades = [];
-  for (const branch of branches) {
-    for (const upgrade of branch.upgrades) {
+  for (const branchName of Object.keys(branchUpgrades)) {
+    for (const upgrade of branchUpgrades[branchName]) {
```
Comment (viceice) [inline lib/workers/repository/changelog/index.ts:13]: Unrelated to this pr, can you extract this please.

Comment (mikebryant) [reply lib/workers/repository/changelog/index.ts:13]: It's not unrelated, all of the changes are required in order to solve #6337 - as discussed there, the key both needs to be allowed in the template, and the point at which the field is generated needs to be changed. Currently it's generated after the template is already compiled.

I'll create two new PRs for the separate pieces, but neither one will solve the issue on their own

Comment (viceice) [reply lib/workers/repository/changelog/index.ts:13]: ok, sorry, i see it's because the changed input. You can leave it here.

**Review by viceice — APPROVED** [review]

File: lib/workers/repository/changelog/index.ts (hunk 1 above)
Comment (mikebryant) [reply lib/workers/repository/changelog/index.ts:13]: I've created https://github.com/renovatebot/renovate/pull/6346 and https://github.com/renovatebot/renovate/pull/6347

If that's what's preferred :)

**Comment by viceice** [comment]
superseeded by #6346 #6447

**Pull request closed without merging**